blog-content/articles/save-your-future-self-by-encapsulating-test-input-preparation.md

```diff
@@ -0,0 +1,97 @@
+---
+date: 2020-01-07
+summary: Examples of how to reduce boilerplate and brittleness of your test input preparation 
+---
+
+# Tests Brittleness and Verbosity slows you down
+
+In [Testing Makes You Faster Day one](entry/testing-makes-you-faster-day-one) article I claimed that writing tests makes you faster.
+However there are test patterns that most likely will slow you down and you need to be ready to handle them. 
+
+Writing tests for a simple logic with a simple input is a breathe. Take for example a test for a simple calculator class. 
+Two numeric inputs and we are done: 
+
+:include-file: com/example/portfolio/SimpleCalculatorTest.java {title: "SimpleCalculatorTest.java"}
+
+Unfortunately real apps require real domain, and real domain is rarely represented by a couple of numbers.
+
+Let's take an example of a fake `ProfitCalculator`. 
+
+Business explained to us how to calculate a potential profit based on a list of `Transactions`. Let's implement
+and test this crucial piece of our software. 
+
+`Transaction` is a simple data class with a few fields (number of fields is reduced for simplicity).
+
+:include-file: com/example/portfolio/Transaction.java {title: "Transaction.java", readMore: true, readMoreVisibleLines: 20}
+
+Here is the first `ProfitCalculator` implementation in all its glory, ready to be tested.
+
+:include-file: com/example/portfolio/ProfitCalculator.java {title: "ProfitCalculator.java"}
+
+And here is the first, boilerplate version of our test.
+
+:include-file: com/example/portfolio/ProfitCalculatorWithoutEncapsulationTest.java {title: "ProfitCalculatorTest.java"}
+
+You may think "test is not that bad", only a few setters to call, not a big deal. 
+The number of setters will only grow as you dig more into requirements. 
+The number of transactions will grow as we deal with more complex scenarios. 
+But that's not all the reasons to be worried.
+
+Another reason to worry is the way `Transaction` instance is being constructed. Right now it is a bunch of setters.
+Tomorrow it becomes a builder pattern, or straight constructor initialization.  
```

Comment:
I would replace *it becomes* with *it could become*.
Also, what do you mean by *straight constructor initialization*? I feel like you are referring to something standard, but I don't know what it is...